src/Lifecycle.js

```diff
@@ -380,10 +380,21 @@ export function startMatrixClient() {
  */
 export function onLoggedOut() {
     _clearLocalStorage();
+    _clearIndexedDB();
     stopMatrixClient();
     dis.dispatch({action: 'on_logged_out'});
 }
 
+function _clearIndexedDB() {
+    // remove indexeddb instances
+    if (!window.indexedDB) {
+        return;
+    }
+    console.log("Clearing indexeddb");
+    window.indexedDB.deleteDatabase("matrix-js-sdk");
```

Comment:
Done. https://github.com/matrix-org/matrix-js-sdk/pull/363/commits/f8d2661426c607a83432edda82adeff7e9c1ac16 and https://github.com/matrix-org/matrix-react-sdk/pull/687/commits/cca266c62cea505d8e80e5e32cac07bfccc1caf3